src/ui/proxy/interceptors.go

```diff
@@ -120,32 +127,18 @@ func (uh urlHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 			http.Error(rw, fmt.Sprintf("Bad repository name: %s", repository), http.StatusBadRequest)
 			return
 		}
-		/*
-			//Need to get digest of the image.
-			endpoint, err := config.RegistryURL()
-			if err != nil {
-				log.Errorf("Error getting Registry URL: %v", err)
-				http.Error(rw, fmt.Sprintf("Failed due to internal Error: %v", err), http.StatusInternalError)
-				return
-			}
-			rc, err := api.NewRepositoryClient(endpoint, false, username, repository, "repository", repository, "pull")
-			if err != nil {
-				log.Errorf("Error creating repository Client: %v", err)
-				http.Error(rw, fmt.Sprintf("Failed due to internal Error: %v", err), http.StatusInternalError)
-				return
-			}
-			digest, exist, err := rc.ManifestExist(tag)
-			if err != nil {
-				log.Errorf("Failed to get digest for tag: %s, error: %v", tag, err)
-				http.Error(rw, fmt.Sprintf("Failed due to internal Error: %v", err), http.StatusInternalError)
-				return
-			}
-		*/
-
+		rec := httptest.NewRecorder()
+		uh.next.ServeHTTP(rec, req)
+		if rec.Result().StatusCode != http.StatusOK {
+			copyResp(rec, rw)
+			return
+		}
+		digest := rec.Header().Get(http.CanonicalHeaderKey("Docker-Content-Digest"))
```

Comment:
Let's discuss, I maybe wrong, but I don't think you should put this in the head of the interceptors.